### DISCUSSION TITLE
Error: Middle policy in Spatial Sorting

### DISCUSSION THREAD

**Discussion by sherlockLitao** [CGAL/cgal#6464] [Q&A] [answered]
Hello,
I have a question about Spatial Sorting.
When I change policy to middle in `Spatial_sorting/sp_sort_using_property_map_d.cpp`, I get something wrong like this:
```
 error: cannot convert ‘CGAL::Null_functor::result_type’ {aka ‘CGAL::Null_tag’} to ‘__gnu_cxx::__alloc_traits<std::allocator<double>, double>::value_type’ {aka ‘double’} in assignment
  152 |  mini[i]=maxi[i]=to_double( _k.compute_coordinate_d_object() (*begin,i) );
      |                  ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                           |
      |                           CGAL::Null_functor::result_type {aka CGAL::Null_tag}
```
But median policy is OK.
I don't why. Could anyone give me some help?
Thanks!

**Comment by sloriot** [comment, marked as answer]
Thanks for the report. The code was buggy. It is fixed in #6465

**Reply by sherlockLitao** [reply]
Thank you very much! This reply is really important to me!
I run the modified code and it works now. I test the code carefully!

However, I found a small bug in Spatial_sorting/include/CGAL/Hilbert_sort_middle_2.h.
If you change the code in Spatial_sorting/examples/Spatial_sorting/sp_sort_using_property_map_3.cpp from dimension 3 to dimension 2,  that is, Point_3 ->Point_2, you will get an error "segmentation fault" when use middle policy!

Here, I change the code in Spatial_sorting/include/CGAL/Hilbert_sort_middle_2.h like Spatial_sorting/include/CGAL/Hilbert_sort_middle_3.h, that is, from "k" to "_k".
For example, 
double xmin=to_double(k.compute_x_3_object()(*begin)),
             ymin=to_double(k.compute_y_3_object()(*begin)),
--->
double xmin=to_double(_k.compute_x_3_object()(*begin)),
             ymin=to_double(_k.compute_y_3_object()(*begin)),

This bug is fixed!

**Reply by sloriot** [reply]
I indeed missed that file. Fixed in #6485
